scalajslib/test/src/HelloSplitJSWorldTests.scala

```diff
@@ -0,0 +1,362 @@
+//        skipScala = _.startsWith("2.11.")
+//      )
+//      testAllMatrix(
+//        (scala, scalaJS, _) => checkScalaTest(scala, scalaJS, cached),
+//        skipScala = isScala3
+//      )
+//    }
+//
+//    "testCached" - {
+//      val cached = false
+//      testAllMatrix(
+//        (scala, scalaJS, _) => checkUtest(scala, scalaJS, cached),
+//        skipScala = _.startsWith("2.11.")
+//      )
+//      testAllMatrix(
+//        (scala, scalaJS, _) => checkScalaTest(scala, scalaJS, cached),
+//        skipScala = isScala3
+//      )
+//    }
+
+//    def checkRun(scalaVersion: String, scalaJSVersion: String, useECMAScript2015: Boolean): Unit = {
+//      val task = HelloJSWorld.helloJsWorld(scalaVersion, scalaJSVersion, useECMAScript2015).run()
+//
+//      val Right((_, evalCount)) = helloWorldEvaluator(task)
+//
+//      val paths = EvaluatorPaths.resolveDestPaths(helloWorldEvaluator.outPath, task)
+//      val log = os.read(paths.log)
+//      assert(
+//        evalCount > 0,
+//        log.contains("node"),
+//        // In Scala.js 1.x, println's are sent to the stdout, not to the logger
+//        !scalaJSVersion.startsWith("0.6.") || log.contains("Scala.js")
+//      )
+//    }
+
+//    "run" - {
+//      testAllMatrix(
+//        (scala, scalaJS, useECMAScript2015) => checkRun(scala, scalaJS, useECMAScript2015),
+//        skipECMAScript2015 = false
+//      )
+//    }
+  }
+
+  def compileClassfiles(parentDir: os.Path, scalaVersion: String, scalaJSVersion: String) = {
+    val inAllVersions = Set(
+      parentDir / "MainA$.class",
+      parentDir / "MainA$.sjsir",
+      parentDir / "MainA.class",
+      parentDir / "MainA.sjsir",
+      parentDir / "MainB$.class",
+      parentDir / "MainB$.sjsir",
+      parentDir / "MainB.class",
+      parentDir / "MainB.sjsir"
+    )
+    val scalaVersionSpecific =
+      if (isScala3(scalaVersion)) Set(
+        parentDir / "ArgsParser.tasty",
+        parentDir / "Main.tasty"
+      )
+      else Set(
+        parentDir / "MainA$delayedInit$body.class",
+        parentDir / "MainA$delayedInit$body.sjsir"
+      )
+    inAllVersions ++ scalaVersionSpecific
+  }
+
+  def prepareWorkspace(): Unit = {
+    os.remove.all(workspacePath)
+    os.makeDir.all(workspacePath / os.up)
+    os.copy(millSourcePath, workspacePath)
+  }
+
+  def testAllMatrix(
+      f: (String, String, Boolean) => Unit,
+      skipScala: String => Boolean = _ => false,
+      skipScalaJS: String => Boolean = _ => false,
+      skipECMAScript2015: Boolean = true
+  ): Unit = {
+    for {
+      (scala, scalaJS, useECMAScript2015) <- HelloJSWorld.matrix
+      if !skipScala(scala)
+      if !skipScalaJS(scalaJS)
+      if !(skipECMAScript2015 && useECMAScript2015)
+    } {
+      if (scala.startsWith("2.11.")) {
+        TestUtil.disableInJava9OrAbove(f(scala, scalaJS, useECMAScript2015))
+      } else {
+        f(scala, scalaJS, useECMAScript2015)
+      }
+    }
+  }
```

Comment:
There is a lot of copy paste in this test. Can you add a minimal test for fastLinkJS avoiding to copy paste too much?